build_tools/build_libtorch.sh

```diff
@@ -102,19 +104,39 @@ fi
 
 checkout_pytorch() {
   if [[ ! -d "$PYTORCH_ROOT" ]]; then
-    git clone https://github.com/pytorch/pytorch "$PYTORCH_ROOT"
+    git clone --depth 1 --single-branch --branch "${PYTORCH_BRANCH}" https://github.com/pytorch/pytorch "$PYTORCH_ROOT"
   fi
   cd "$PYTORCH_ROOT"
-  git fetch --all
-  git checkout origin/"${PYTORCH_BRANCH}"
-  git submodule update --init --recursive
+  for dep in protobuf pocketfft cpuinfo FP16 psimd fmt sleef pybind11 onnx flatbuffers foxi; do
+    git submodule update --init --depth 1 -- third_party/$dep
+  done
 }
 
 build_pytorch() {
   cd "$PYTORCH_ROOT"
   # Uncomment the next line if you want to iterate on source builds
-  python setup.py clean
-
+  # python setup.py clean
+
+  sed -i.bak -E 's/^[[:space:]]+check_submodules()/#check_submodules()/g' setup.py
+  # remove after https://github.com/pytorch/pytorch/issues/81178 is fixed
```

Comment:
dirty dirty hack but necessary until https://github.com/pytorch/pytorch/issues/81178 is resolved